### PR TITLE
Let move_file API support moving of both files and folders and deprecate move_tree API

### PR DESCRIPTION
**Pull request by TeusEE**
after https://github.com/lablup/backend.ai-storage-proxy/issues/11#issue-743489261

move_tree are added in vfs/init,py

change function(move file) can handle both file and folder

### CODE REVIEW

**Comment by lizable** [comment]
@TeusEE please add a news fragment according to the [link🔗](https://github.com/lablup/contribution-academy-2021/discussions/91).

**Comment by lizable** [comment]
@TeusEE And I wonder you checked whether the main problem(cannot rename directory in the virtual folder) has been solved or not. I checked it and It seems that I can rename the directory in the vfolder without any error. Please see the image below.

directory-renaming

**Comment by TeusEE** [comment]
> @TeusEE And I wonder you checked whether the main problem(cannot rename directory in the virtual folder) has been solved or not. I checked it and It seems that I can rename the directory in the vfolder without any error. Please see the image below.
> 
> directory-renaming

In CLI, move file can't handle folder. maybe WEB UI Use move folder and move file both i guess

File: src/ai/backend/storage/vfs/__init__.py [hunk 1]
```diff
@@ -282,18 +282,26 @@ async def move_file(
...
+            await loop.run_in_executor(None, src_path.rename, dst_path)
+        else:
+            raise InvalidAPIParameters(
+                msg=f"source path {str(src_path)} is not a file or dir"
+            )
 
     async def move_tree(
         self, vfid: UUID, src: PurePosixPath, dst: PurePosixPath
     ) -> None:
+        print("Use move_file instead. move tree will be deprecated")
```
Comment (achimnol) [inline src/ai/backend/storage/vfs/__init__.py:304]: If you want to deprecate an API, you should import and use `warnings` like:
```suggestion
        warnings.warn("Use move_file instead. move tree will be deprecated", DeprecationWarning, stacklevel=2)
```

Comment (achimnol) [reply src/ai/backend/storage/vfs/__init__.py:304]: The reason to separate file moves and directory moves was to distinguish any vendor-specific optimization implementations, but I think we could just do it inside the `move_file()`.

Comment (achimnol) [reply src/ai/backend/storage/vfs/__init__.py:304]: The deprecation notice should also be in the abstract base class as a comment as well.

**Review by TeusEE — COMMENTED** [review]
@achimnol Apply your recommendation complete.
change inform Deprecate API method print->warning
PLZ Review PR again

**Comment by achimnol** [comment]
First fix up lint/typecheck errors. Maybe due to missing import of warnings.

**Comment by TeusEE** [comment]
@achimnol amend lint and misc error. PLZ Review PR again

**Pull request merged**